nntrainer/layers/mol_attention_layer.cpp

```diff
@@ -96,59 +95,59 @@ void MoLAttentionLayer::finalize(InitLayerContext &context) {
   auto &bias_decay = std::get<props::BiasDecay>(*layer_impl_props);
 
   TensorDim fc_w_dim = {query_dim.width(), unit};
-  wt_idx[AttentionParams::fc_w] = context.requestWeight(
+  wt_idx[MoLAttentionParams::fc_w] = context.requestWeight(
     fc_w_dim, weight_initializer, weight_regularizer,
     weight_regularizer_constant, weight_decay, "fc_w", true);
   TensorDim fc_bias_dim = {unit};
-  wt_idx[AttentionParams::fc_bias] = context.requestWeight(
+  wt_idx[MoLAttentionParams::fc_bias] = context.requestWeight(
     fc_bias_dim, bias_initializer, weight_regularizer,
     weight_regularizer_constant, bias_decay, "fc_bias", true);
 
   TensorDim fc_proj_w_dim = {unit, 3 * mol_k};
-  wt_idx[AttentionParams::fc_proj_w] = context.requestWeight(
+  wt_idx[MoLAttentionParams::fc_proj_w] = context.requestWeight(
     fc_proj_w_dim, weight_initializer, weight_regularizer,
     weight_regularizer_constant, weight_decay, "fc_proj_w", true);
 
   TensorDim fc_out_dim = query_dim;
   fc_out_dim.width(fc_w_dim.width());
-  wt_idx[AttentionParams::fc_out] =
+  wt_idx[MoLAttentionParams::fc_out] =
     context.requestTensor(fc_out_dim, "fc_out", Tensor::Initializer::NONE,
-                          false, TensorLifespan::ITERATION_LIFESPAN);
+                          false, TensorLifespan::FORWARD_FUNC_LIFESPAN);
```

Comment:
Isn't it used for the backwarding?

Reply:
In the backwarding only the dimension of fc_out is used so I revised it.